lib/shopify-cli/tasks/ensure_env.rb

```diff
@@ -3,27 +3,85 @@
 module ShopifyCli
   module Tasks
     class EnsureEnv < ShopifyCli::Task
-      def call(ctx)
+      def call(ctx, regenerate: false)
         @ctx = ctx
-        Resources::EnvFile.read(ctx.root)
+        env_data = Resources::EnvFile.parse_external_env
+        if regenerate
+          write_env(ask(env_data))
+        else
+          write_env(ask(env_data)) unless [:api_key, :secret, :shop].all? { |property| env_data[property] }
+        end
       rescue Errno::ENOENT
-        ask
+        write_env(ask)
```

Comment:
If you are just going to pass in ask for every invocation of `write_env` then either `ask` should handle writing or `write_env` should handle asking. Also the branching in this is out of sorts.
```ruby
write_env(ask(env_data)) if regenerate || ![:api_key, :secret, :shop].all? { |property| env_data[property] }
```
Also maybe you should pass `[:api_key, :secret, :shop]` as an argument into the call? It seems arbitrary for your project to need because apps need scopes.